src/Symfony/Component/Console/Application.php

```diff
@@ -940,7 +940,7 @@ protected function doRunCommand(Command $command, InputInterface $input, OutputI
             }
         }
 
-        if ($command instanceof SignalableCommandInterface) {
+        if ($command instanceof SignalableCommandInterface && !(empty($this->signalsToDispatchEvent) && empty($command->getSubscribedSignals()))) {
```

Comment:
Symfony core does not use `empty()`.

```suggestion
        if ($command instanceof SignalableCommandInterface && ($this->signalsToDispatchEvent || $command->getSubscribedSignals())) {
```

Reply:
note that the suggested replacement is not correct. the negation of a AND is a OR, not a AND.

Reply:
right, fixed